### PR TITLE
feat: graphing util untested

### LINKED ISSUE #136: 🚀 Feature: Graphing Util for A3 Game (used for slide 33+35)
util for graphing like [this slide](https://www.figma.com/file/7DzNiScbp6t3x0p86pz5P9/Playnets?node-id=1846%3A77870)

* [ ] :wave: I may be able to implement this feature request

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by reginawang99**
closes #136 

EDIT:
- brought reset of `setDaysLeft` from `TimeAllocation.tsx` to GameManager
- added typing of `timeAllocations`  and `Point`
    - so that they are easier to use and require less codebase digging to find semantics!

### CODE REVIEW

File: src/utils/numbers.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @returns squeezed result + change in num
+ */
+export function squeeze(num: number, min: number, max: number) : [number, number]{
```
Comment (BryanPan342) [inline src/utils/numbers.ts:4]: the general term for this is clamp!

Comment (BryanPan342) [reply src/utils/numbers.ts:4]:
```suggestion
export function clamp(min: number, num: number, max: number) : [number, number]{
```

File: src/utils/numbers.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @returns squeezed result + change in num
+ */
```
Comment (BryanPan342) [inline src/utils/numbers.ts:3]:
```suggestion
/**
 * Given a number and its min and max range, return either the number or the edges of the range.
 *
 * @returns [result, difference]
 * @var result: the clamped value
 * @var difference: the difference between the clamped value and the input num
 */
```

File: src/components/Activities/Activity3/Game/GameConstantsToMessWith.ts [hunk 3]
```diff
@@ -12,7 +19,7 @@ export const DEBUG_ERROR_OPTIONS = [
   'Image failed to load', 'File not found', 'Unexpected any.', 'Expected linebreaks to be \'LF\' but found \'CRLF\'',
   '\',\' expected.', 'Cannot find name \'numbe\'.', '\'expectedWeights\' is declared but its value is never read.',
   'All destructured elements are unused.', 'Maximum allowed is 120', 'ReferenceError: MAX_NUM_ERRORS is not defined',
-  'Warning: Function components cannot be given refs.', 'net::ERR_ABORTED 403 (Forbidden)',
+  'Warning: Function components cannot be given refs.', 'net::ERR_ABORTED 403 (Forbidden)', 'React version not specified',
```
Comment (BryanPan342) [inline src/components/Activities/Activity3/Game/GameConstantsToMessWith.ts:22]: LMAO what is this?

File: src/components/Activities/Activity3/Game/gameCalculationsUtil.ts [hunk 4]
```diff
@@ -40,54 +93,82 @@ export function getDebugErrors(
...
+    [lastY, changeInY] = squeeze(lastY + change, 0, 100);
+    xyMapping.push([lastX, lastY]);
+    dxyMapping.push([changeInX, changeInY]);
+  }
+  return { xyMap: xyMapping, dxyMap: dxyMapping };
 }
 
 /**
- *
  * @returns points between x: [0, 100], y: [0, 100] to graph
```
Comment (BryanPan342) [inline src/components/Activities/Activity3/Game/gameCalculationsUtil.ts:143]: if we could add a really general comment on how the function works that would be awesome!

File: src/components/Activities/Activity3/Game/gameCalculationsUtil.ts [hunk 5]
```diff
@@ -40,54 +93,82 @@ export function getDebugErrors(
...
   ) ?? QUALITY_DEFAULT_KEY;
   const castingKey = qualityKey as keyof typeof NUMBER_TO_QUALITY_MAP;
   return NUMBER_TO_QUALITY_MAP[castingKey];
 }
 
 /**
- *
  * @returns points between x: [0, 100], y: [0, 100] to graph
  */
 export function getControlGraphForABTesting(
```
Comment (BryanPan342) [inline src/components/Activities/Activity3/Game/gameCalculationsUtil.ts:121]: if we could add a really general comment on how the function works that would be awesome!

**Review by BryanPan342 — CHANGES_REQUESTED** [review]
i didnt really look into the logic for the functions, but I like the way the return structure is working

if we could just add a single sentence describing the function similar in spirit to the cipher stuff that would be awesome

https://github.com/uclaacm/Playnet/blob/4334c0312940a0229ba10b310c6b9887077eb9b5/src/utils/cipher.ts#L17-L22

File: src/components/Activities/Activity3/Game/GameConstantsToMessWith.ts (hunk 3 above)
Comment (reginawang99) [reply src/components/Activities/Activity3/Game/GameConstantsToMessWith.ts:22]: LOLLL debug screen ahaha 
image

ive just been adding random bugs there ahaha

Comment (BryanPan342) [reply src/components/Activities/Activity3/Game/GameConstantsToMessWith.ts:22]: ooo haha okay sounds good

**Review by BryanPan342 — APPROVED** [review]
Good enough for me!

**Pull request merged**